tests/test_common_args.py

```diff
@@ -0,0 +1,78 @@
+"""
+<Program Name>
+  test_common_args.py
+
+<Author>
+  Lukas Puehringer <lukas.puehringer@nyu.edu>
+
+<Started>
+  Jan 16, 2020
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  Test in_toto/common_args.py
+
+"""
+import unittest
+import argparse
+import in_toto.common_args
+
+
+class TestArgparseActionGroupHelpers(unittest.TestCase):
+  # pylint: disable=protected-access
+
+  def setUp(self):
+    """Create an empty parser and perform some basic assertions prior to
+    testing parser action group (i.e. argument group) helper functions. """
+    # Create empty argument parser
+    self.parser = argparse.ArgumentParser()
+
+    # Assert parser has the protected member "_action_groups" and it is a list
+    # NOTE: argparse could remove this at any time without notice
+    self.assertTrue(
+        isinstance(getattr(self.parser, "_action_groups", None), list))
+
```

Comment:
I like we are testing against this, but I've always been wary of `type` vs `isinstance`. I personally don't know if we want to ensure it's a list-ish type, or must be a list and nothing else. What's your take?

Reply:
You are absolutely right. In this case we should do a `type` and not an `instance` check, because in `common_args.sort_action_groups` we replace what `_action_groups` references with a list.  So if `argparse` were to change the type to something list-ish, it could lead to unexpected behavior and we might not detect it in the test here. Thanks for pointing this out. I'll push a change.